src/reg.js

```diff
@@ -36,3 +36,6 @@ exports.fallsThrough = /^\s*\/\*\s*falls?\sthrough\s*\*\/\s*$/;
 // very conservative rule (eg: only one space between the start of the comment and the first character)
 // to relax the maxlen option
 exports.maxlenException = /^(?:(?:\/\/|\/\*|\*) ?)?[^ ]+$/;
+
+// Unicode escape sequences (\uXXXX and \u{XXXXXX})
+exports.unicodeEscapeSequence = /\\u(?:([0-9a-fA-F]{4})|\{([0-9a-fA-F]{1,6})\})/g;
```

Comment:
This RegExp has a hard-coded 6 from your initial solution. That gets in the way
of tracking references to variables; the following test fails:

    TestRun(test)
      .test("var \\u{00000078};\\u{78} = 9;", { esnext: true, unused: true });

Changing the `{1,6}` to a `+` should fix it, though!